src/main/groovy/pl/allegro/tech/build/axion/release/domain/NextVersionMarker.groovy

```diff
@@ -9,22 +10,34 @@ class NextVersionMarker {
 
     private static final ReleaseLogger logger = ReleaseLogger.Factory.logger(NextVersionMarker)
 
-    private final ScmService repository
+    private final ScmService repositoryService
+    private final VersionConfig versionConfig
 
-    NextVersionMarker(ScmService repository) {
-        this.repository = repository
+    NextVersionMarker(ScmService repositoryService, VersionConfig config) {
+        this.repositoryService = repositoryService
+        this.versionConfig = config
     }
 
     void markNextVersion(NextVersionProperties nextVersionRules, TagProperties tagRules) {
-        if (nextVersionRules.nextVersion == null) {
-            throw new IllegalArgumentException("No next version specified! Use -Prelease.version to set next version.")
+
+        def version = nextVersionRules.nextVersion
+        def currentVersion = Version.valueOf(versionConfig.version)
```

Comment:
I think that you should use undecorated version instead. Decorated version might have been altered in the process (branch name, snapshot added etc).